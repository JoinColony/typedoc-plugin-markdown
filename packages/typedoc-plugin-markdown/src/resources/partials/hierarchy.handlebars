{{#each types}}
{{#if ../isTarget}}
{{{spaces ../level}}}- **`{{this}}`**
{{else}}
{{{spaces ../level}}}- {{{type this}}}{{/if}}
{{#if @last}}
{{#with ../next}}
{{~#iterate this 'level' ../../level~}}
{{~> hierarchy level=level~}}
{{~/iterate~}}
{{~/with~}}
{{~/if~}}
{{~/each~}}