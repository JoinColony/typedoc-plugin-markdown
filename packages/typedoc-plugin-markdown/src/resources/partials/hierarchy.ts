import { DeclarationHierarchy } from 'typedoc/dist/lib/models';
import { MarkdownThemeContext } from '../..';

export const hierarchyPartial = (
  context: MarkdownThemeContext,
  props: DeclarationHierarchy,
  level: number,
) => {
  const md: string[] = [];
  const symbol = level > 0 ? getSymbol(level) : '-';
  props.types.forEach((hierarchyType) => {
    if (props.isTarget) {
      md.push(`${symbol} **\`${hierarchyType}\`**`);
    } else {
      md.push(`${symbol} ${context.typePartial(hierarchyType)}`);
    }
  });
  if (props.next) {
    md.push(context.hierarchyPartial(props.next, level + 1));
  }
  return md.join('\n\n');
};

function getSymbol(level: number) {
  return '  ' + [...Array(level)].map(() => 'â†³').join('');
}
