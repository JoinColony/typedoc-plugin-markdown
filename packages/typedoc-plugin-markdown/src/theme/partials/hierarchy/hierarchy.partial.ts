import { DeclarationHierarchy } from 'typedoc';
import { MarkdownThemeRenderContext } from '../../theme.context';
import { backTicks, bold } from '../../utils/elements';

export const hierarchyPartial = (
  context: MarkdownThemeRenderContext,
  props: DeclarationHierarchy,
) => {
  let level = 0;
  const getHierarchy = (props: DeclarationHierarchy) => {
    return props.types.map((hierarchyType) => {
      level = level + 1;
      return (
        getSymbol(level) +
        (props.isTarget
          ? bold(backTicks(hierarchyType.toString()))
          : context.typePartial(hierarchyType))
      );
    });
  };
  let md = getHierarchy(props);
  if (props.next) {
    md = [...md, ...getHierarchy(props.next)];
  }
  return md.join('\n');
};

const getSymbol = (level: number) => {
  return `${[...Array(level - 1)].map(() => '  ').join('')}- `;
};
