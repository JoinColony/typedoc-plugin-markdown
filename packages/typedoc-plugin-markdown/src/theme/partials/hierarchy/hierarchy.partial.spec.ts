import { DeclarationReflection, ProjectReflection } from 'typedoc';
import { hierarchyPartial } from './hierarchy.partial';

describe(`Hierarchy Partial:`, () => {
  let project: ProjectReflection;

  beforeAll(async () => {
    project = global.getProject(__dirname);
  });

  test(`should compile type hierarchy`, () => {
    expect(
      hierarchyPartial(
        global.getMockContext(),
        (project.findReflectionByName('ParentClass') as DeclarationReflection)
          .typeHierarchy,
      ),
    ).toMatchSnapshot();
  });
});
