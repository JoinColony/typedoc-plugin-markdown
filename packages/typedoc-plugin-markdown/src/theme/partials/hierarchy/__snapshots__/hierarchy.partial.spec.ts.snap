// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hierarchy Partial: should compile type hierarchy 1`] = `
"- **\`ParentClass\`**
  - [\`ChildClassA\`]( ChildClassA )
    - [\`ChildClassB\`]( ChildClassB )"
`;
